test/parallel/test-timers-unrefd-interval-still-fires.js

```diff
@@ -1,26 +1,25 @@
 'use strict';
 /*
  * This test is a regression test for joyent/node#8900.
+ *
+ * Node.js 0.10.34 contains the bug that this test detects.
+ * Node.js 0.10.35 contains the fix.
+ *
+ * Due to the introduction of ES6-isms, comment out `'use strict'` and
+ * `require('../common');` to run the test on Node.js 0.10.x.
  */
-const common = require('../common');
+require('../common');
 
-const TEST_DURATION = common.platformTimeout(100);
 const N = 3;
 var nbIntervalFired = 0;
 
-const keepOpen = setTimeout(() => {
-  console.error('[FAIL] Interval fired %d/%d times.', nbIntervalFired, N);
-  throw new Error('Test timed out. keepOpen was not canceled.');
-}, TEST_DURATION);
+const keepOpen = setInterval(function() {}, 9999);
 
-const timer = setInterval(() => {
+const timer = setInterval(function() {
   ++nbIntervalFired;
   if (nbIntervalFired === N) {
     clearInterval(timer);
-    timer._onTimeout = () => {
-      throw new Error('Unrefd interval fired after being cleared.');
-    };
```

Comment:
You're not wrong, but at the same time:

* Without that code, the test times out if the `clearInterval()` somehow doesn't work. So it's not strictly necessary. 

* Presumably other tests check to confirm that `clearInterval()` works, so tacking it on to this test is kinda sorta scope creep. 

* If the test is for one thing and one thing only, it's a whole lot easier to figure out what's wrong when the test fails. 

* Minor point, but the access to a `_`-prefixed property is kind of a code smell.

But you're not wrong on your point that the removed code provides a proper error if `clearInterval()` does somehow fail. Happy to put it back in if it's a deal-breaker.